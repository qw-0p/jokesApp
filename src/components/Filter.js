import React from 'react';
import { DropdownButton, Dropdown, Col } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { filterList } from '../redux/ducks/jokes';

export const Filter = () => {
    const dispatch = useDispatch();
    const state = useSelector((state) => state.jokes.categories);
    const loading = useSelector((state) => state.jokes.loading);
    const submitHandler = (category) => {
        dispatch(filterList(category));
    };

    return (
        <Col>
            <DropdownButton id='dropdown-item-button' title='Filter' disabled={loading}>
                <Dropdown.Item as='button' onClick={() => submitHandler(null)}>
                    all
                </Dropdown.Item>
                {state.map(
                    (item) =>
                        item && (
                            <Dropdown.Item as='button' key={item} onClick={() => submitHandler(item)}>
                                {item}
                            </Dropdown.Item>
                        )
                )}
            </DropdownButton>
        </Col>
    );
};
