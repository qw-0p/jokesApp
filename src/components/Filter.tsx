import React from 'react';
import { DropdownButton, Dropdown, Col } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { filterList } from '../redux/ducks/jokes';
import { RootState } from '../redux/store';


export const Filter: React.FC = () => {
    const dispatch = useDispatch();
    const state = useSelector((state: RootState): string[] => state.jokes.categories);
    const loading = useSelector((state: RootState): boolean => state.app.loading);
    const submitHandler = (event: any): void => {
        dispatch(filterList(event));
    };
    
    return (
        <Col>
            <DropdownButton id='dropdown-item-button' title='Filter' disabled={loading}>
                <Dropdown.Item as='button' onClick={(): void => submitHandler(null)}>
                    all
                </Dropdown.Item>
                {state.map(
                    (item: any): JSX.Element =>
                        item && (
                            <Dropdown.Item as='button' key={item} onClick={(): void => submitHandler(item)}>
                                {item}
                            </Dropdown.Item>
                        )
                )}
            </DropdownButton>
        </Col>
    );
};
