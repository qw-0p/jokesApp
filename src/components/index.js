export { Error } from './Error';
export { Filter } from './Filter';
export { Home } from './Home';
export { Item } from './Item';
export { JokesForm } from './JokesForm';
export { List } from './List';
export { Loader } from './Loader';
