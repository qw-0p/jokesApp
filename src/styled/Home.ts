import { Container } from 'react-bootstrap';
import styled from 'styled-components';

export const Wrapper = styled(Container)`
    margin: 25px 0;
`;
