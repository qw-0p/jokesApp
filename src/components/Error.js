import React from 'react';
import { Row } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { ErrorPage } from '../styled/Error';

export const Error = () => (
    <ErrorPage>
        <Row>Page not found</Row>
        <Row>
            <Link to='/'>Home</Link>
        </Row>
    </ErrorPage>
);
