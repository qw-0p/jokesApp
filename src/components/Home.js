import React from 'react';
import { Col, Row } from 'react-bootstrap';
import { JokesForm } from './JokesForm';
import { Filter } from './Filter';
import { List } from './List';
import { Loader } from './Loader';
import { Wrapper } from '../styled/Home';

export const Home = () => (
    <Wrapper fluid>
        <Row>
            <Col>
                <Filter />
            </Col>
        </Row>
        <Row>
            <Col>
                <List />
                <Loader />
            </Col>
        </Row>
        <Row className='justify-content-center'>
            <JokesForm />
        </Row>
    </Wrapper>
);
