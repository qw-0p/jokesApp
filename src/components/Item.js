import React, { useState } from 'react';
import { Row, Modal, Button } from 'react-bootstrap';
import { Link, useHistory, useParams, useLocation } from 'react-router-dom';
import { JokePage } from '../styled/Item';

export const Item = (props) => {
    let location = useLocation();
    let history = useHistory();
    let { id } = useParams();
    // console.log('History', history);
    // console.log('Params', id);
    // console.log('Loc', location.state);
    let back = (e) => {
        e.stopPropagation();
        history.goBack();
    };
    return (
        <Modal {...props}>
            <Modal.Body>
                <p>{location.state}</p>
            </Modal.Body>
            <Modal.Footer>
                <Button onClick={back}>Close</Button>
            </Modal.Footer>
        </Modal>
    );
};
