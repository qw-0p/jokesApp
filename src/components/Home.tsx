import React from 'react';
import { Col, Row } from 'react-bootstrap';

import { JokesForm, Filter, List, Loader } from '.';
import { Wrapper } from '../styled/Home';

export const Home: React.FC = () => (
    <Wrapper fluid>
        <Row>
            <Col>
                <Filter />
            </Col>
        </Row>
        <Row>
            <Col>
                <List />
                <Loader />
            </Col>
        </Row>
        <Row className='justify-content-center'>
            <JokesForm />
        </Row>
    </Wrapper>
);
