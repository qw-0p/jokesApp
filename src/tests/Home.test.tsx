import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Home } from '../components/Home';

describe('Home component', () => {
    it('snapshot', () => {
        const component = shallow( <Home />);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});