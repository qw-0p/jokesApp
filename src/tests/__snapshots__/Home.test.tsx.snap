// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component snapshot 1`] = `
<Styled(Container)
  fluid={true}
>
  <Row
    noGutters={false}
  >
    <Col>
      <Filter />
    </Col>
  </Row>
  <Row
    noGutters={false}
  >
    <Col>
      <List />
      <Loader />
    </Col>
  </Row>
  <Row
    className="justify-content-center"
    noGutters={false}
  >
    <JokesForm />
  </Row>
</Styled(Container)>
`;
